### ISSUE TITLE
error importing photutils in python3

### ISSUE THREAD

**Issue by gvdsteene** [astropy/photutils#1021]
python3 astropy and photutils are installed with pip3 and up to date
To test it I go in python3 and try to import photutils to run the test, 
but I get an error in the satrapy modelling package.
What is up, please ?

>> import photutils
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python3.7/site-packages/photutils/__init__.py", line 36, in <module>
    from .centroids import *  # noqa
  File "/usr/local/lib/python3.7/site-packages/photutils/centroids/__init__.py", line 7, in <module>
    from .core import *    # noqa
  File "/usr/local/lib/python3.7/site-packages/photutils/centroids/core.py", line 25, in <module>
    class _GaussianConst1D(Const1D + Gaussian1D):
  File "/usr/local/lib/python3.7/site-packages/astropy/modeling/core.py", line 59, in <lambda>
    return lambda left, right: CompoundModel(oper, left, right, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/astropy/modeling/core.py", line 2357, in __init__
    if op != 'fix_inputs' and len(left) != len(right):
TypeError: object of type '_ModelMeta' has no len()

**Comment by bsipocz** [comment]
Could you copy paste what `pip3 freeze` returns to you?

**Comment by rjavila** [comment]
I am seeing the same error from the latest astroconda installation.

**Comment by bsipocz** [comment]
what versions of astropy, photutils and numpy do you have?

**Comment by rjavila** [comment]
Actually, I only thought it was the latest version. I had recently ran a `condo updagrade --all` and thought photutils was up to date, but I had version 0.6. I upgraded to 0.7.2 and things run fine now.

**Comment by gvdsteene** [comment]
I am using it on ubuntu, but I have trouble on Mac
on mac 10.15.3
python3:
numpy==1.18.1
astropy==4.0
photutils==0.7.2

**Comment by ferrigno** [comment]
Hi, I am trying to use it in a docker with python 3.6.9, but I got this error



ipython
/pyenv/versions/3.6.9/lib/python3.6/site-packages/IPython/paths.py:60: UserWarning: IPython dir '/home/jovyan/.ipython' is not a writable location, using a temp directory.
  " using a temp directory.".format(ipdir))
Python 3.6.9 (default, Apr 21 2020, 19:36:51) 
Type 'copyright', 'credits' or 'license' for more information
IPython 7.13.0 -- An enhanced Interactive Python. Type '?' for help.

In [1]: import photutils                                                                                                                                   
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-1-e3d8b5fe96af> in <module>
----> 1 import photutils

/pyenv/versions/3.6.9/lib/python3.6/site-packages/photutils/__init__.py in <module>
     17 
     18 if not _ASTROPY_SETUP_:  # noqa
---> 19     from .aperture import *  # noqa
     20     from .background import *  # noqa
     21     from .centroids import *  # noqa

/pyenv/versions/3.6.9/lib/python3.6/site-packages/photutils/aperture/__init__.py in <module>
      5 
      6 from .bounding_box import *  # noqa
----> 7 from .circle import *  # noqa
      8 from .core import *  # noqa
      9 from .ellipse import *  # noqa

/pyenv/versions/3.6.9/lib/python3.6/site-packages/photutils/aperture/circle.py in <module>
     11 from .attributes import (AngleOrPixelScalarQuantity, PixelPositions,
     12                          PositiveScalar, SkyCoordPositions)
---> 13 from .core import PixelAperture, SkyAperture
     14 from .mask import ApertureMask
     15 from ..geometry import circular_overlap_grid

/pyenv/versions/3.6.9/lib/python3.6/site-packages/photutils/aperture/core.py in <module>
    102 
    103 
--> 104 class PixelAperture(Aperture):
    105     """
    106     Abstract base class for apertures defined in pixel coordinates.

/pyenv/versions/3.6.9/lib/python3.6/site-packages/photutils/aperture/core.py in PixelAperture()
    367     @deprecated_renamed_argument('unit', None, '0.7')
    368     def do_photometry(self, data, error=None, mask=None, method='exact',
--> 369                       subpixels=5, unit=None):
    370         """
    371         Perform aperture photometry on the input data.

/pyenv/versions/3.6.9/lib/python3.6/site-packages/astropy/utils/decorators.py in decorator(function)
    477                                 'signature. If it was meant to be part of '
    478                                 '"**kwargs" then set "arg_in_kwargs" to "True"'
--> 479                                 '.'.format(new_name[i]))
    480 
    481         @functools.wraps(function)

TypeError: "None" was not specified in the function signature. If it was meant to be part of "**kwargs" then set "arg_in_kwargs" to "True".

photutils has
version = "0.7.2"
githash = "885aee002153269aee15c5923c662c48ad132532"

**Comment by larrybradley** [comment]
@ferrigno  This appears to be due to an incompatible astropy version.  The astropy `deprecated_renamed_argument` decorator in `photutils 0.7.2` requires `astropy >= 2.0.12`, but `not 3.0.*, 3.1, or 3.1.1`.  The documentation doesn't mention the exclusions, but that will be fixed in the next release.  Running `pip check` in your environment should report any incompatible dependencies.

**Comment by ferrigno** [comment]
@larrybradley thank you very much, installing astropy 4 in the docker solves my problem.

**Comment by bsipocz** [comment]
@larrybradley - oh, that's a good lesson learnt, we need to keep an eye for this for in the other packages and LTS releases.

**Comment by larrybradley** [comment]
@gvdsteene I've investigated your issue and determined that it's also due to a mismatch in `astropy` and `photutils` required versions.  From your traceback, I see:

 `File "/usr/local/lib/python3.7/site-packages/photutils/centroids/core.py", line 25, in
class _GaussianConst1D(Const1D + Gaussian1D):`

But that line was removed in https://github.com/astropy/photutils/pull/824 in `photutils v0.7`.  Thus, you must actually have `photutils < 0.7` with `astropy 4.0`.

I've confirmed that I get the exact error as you with `photutils 0.6` and `astropy 4.0` installed.  Those versions are incompatible.  If you are using `pip`, `pip check` will report any inconsistencies with your package dependencies.
